src/test/java/org/folio/rest/support/kafka/FakeKafkaConsumer.java

```diff
@@ -133,36 +133,19 @@ private static KafkaConsumerRecord<String, JsonObject>  getLastEvent(
     // The testing paradigm needs to account for the asynchronous nature rather
     // than assuming the "first" or "last" event represent the expected
     // response.
-    Iterator<KafkaConsumerRecord<String, JsonObject>> iter = events.stream().iterator();
-    KafkaConsumerRecord<String, JsonObject> last = null;
-
-    while (iter.hasNext()) {
-      KafkaConsumerRecord<String, JsonObject> record = iter.next();
-
-      if (last == null || record.timestamp() > last.timestamp()) {
-        last = record;
-      }
-    }
 
-    return last;
+    return events.stream()
+      .max(Comparator.comparing(KafkaConsumerRecord::timestamp))
+      .orElse(null);
   }
 
   private static KafkaConsumerRecord<String, JsonObject>  getFirstEvent(
     Collection<KafkaConsumerRecord<String, JsonObject> > events) {
 
     // See also the comment in getLastEvent() above.
-    Iterator<KafkaConsumerRecord<String, JsonObject>> iter = events.stream().iterator();
-    KafkaConsumerRecord<String, JsonObject> first = null;
-
-    while (iter.hasNext()) {
-      KafkaConsumerRecord<String, JsonObject> record = iter.next();
-
-      if (first == null || record.timestamp() < first.timestamp()) {
-        first = record;
-      }
-    }
-
-    return first;
+    return events.stream()
+      .min(Comparator.comparing(KafkaConsumerRecord::timestamp))
+      .orElse(null);
```

Comment:
I'm happy to rename the method. I was trying to re-implement it whilst preserving the intended behaviour, and thus didn't want to rename it at the same time.

Min sorts in ascending order.

I used `min` instead of `sorted` and `first` when IntelliJ suggested it was equivalent.

I could move used `sorted` and `first` if that could be more intent revealing?

Reply:
Hmm. that's a good point.  I think leave the name as it is, then.  This may be just me, I always get a bit thrown when we talk about "min" "max" or "greater" or "less" in terms of time values.

Reply:
> This may be just me, I always get a bit thrown when we talk about "min" "max" or "greater" or "less" in terms of time values.

Would using `sorted` and `first` be more expressive than using `min`?

The timestamp in these contexts are modelled as a long. I presume it's an offset from the epoch.

I was confused by this because I expected it to be a string representation of a date and time.

Reply:
Yes, I think sorted and first are more expressive.